build_scripts/windows/Product.wxs

```diff
@@ -8,18 +8,57 @@
   <?define ProductAuthor = "Microsoft Corporation" ?>
   <?define ProductResources = ".\resources\" ?>
 
+  <?if $(var.Platform) = "x64" ?>
+  <?define AzureCliRegistryGuid = "90b6367e-4e15-4313-af33-a0ef459a56fc" ?>
+  <?define AzureCliSystemPathGuid = "12c8f0dc-0475-4814-93b6-fcbe82e89532" ?>
+  <?define AzureCliVersionGuid = "4df4882f-4604-4cdc-a0b4-bffc4b8c5967" ?>
+  <?define ProgramFilesFolder = "ProgramFiles64Folder" ?>
+  <?define RemoveAzureCLIFolderGuid = "7ce8a521-7429-4ded-9cce-0948e393cc83" ?>
+  <?define RemoveCLIFolderGuid = "59fb416c-91d0-49aa-a185-d4330c435a93" ?>
+  <?define UpgradeCode = "90762fec-9554-4729-a107-c6a8ea316698" ?>
+
+  <?elseif $(var.Platform) = "x86" ?>
+  <?define AzureCliRegistryGuid = "BDEEE50E-70D3-4990-BFF9-FCF8114AA701" ?>
+  <?define AzureCliSystemPathGuid = "3B60CA67-DB92-465A-BC96-3CA6CAEFF41B" ?>
+  <?define AzureCliVersionGuid = "A5647E93-02B7-4CED-802A-DDD7416E0231" ?>
+  <?define ProgramFilesFolder = "ProgramFilesFolder" ?>
+  <?define RemoveAzureCLIFolderGuid = "6AF97701-6A7A-4292-95D6-ED34CD4C97C0" ?>
+  <?define RemoveCLIFolderGuid = "24DA394D-B03D-4623-A0D9-852DB51EF9F4" ?>
+  <?define UpgradeCode = "dff82af0-3f95-4ac9-8efd-948604fdb028" ?>
+
+  <?else ?>
+  <?error Unsupported platform "$(var.Platform)" ?>
+  <?endif ?>
+
   <Product Id="*"
            Name="$(var.ProductName)"
```

Comment:
Some programs attach `x86` and `x64` to the product name. Some don't:

![image](https://github.com/Azure/azure-cli/assets/4003950/a41742b3-18e2-425d-8f9e-374212fd9a9d)

1. Shall we do the same and change `$(var.ProductName)` too?
2. Shall we allow 32-bit and 64-bit MSIs to be installed side-by-side? I don't feel this is necessary as there can be only one active `az.cmd` on `PATH`.

Reply:
> Shall we do the same and change $(var.ProductName) too?

Sure I can do that.

> Shall we allow 32-bit and 64-bit MSIs to be installed side-by-side? I don't feel this is necessary as there can be only one active az.cmd on PATH.

I don't recommend it. 1) what purpose does it serve? These are EXEs where, unlike libraries e.g., DLLs, architecture doesn't typically matter. 32- and 64-bit code can call any architecture of an EXE. It would be a UX problem because,

1) Depending on the ordering of updates to `PATH`, customer will call the first or last architecture installed. It's not deterministic.
2) Every time there's an update, the ~15-20 minute upgrade of 1 MSI will now be doubled.

And all that for what reason? Again, it's an EXE (well, effectively - an executable, more specifically) so architecture doesn't matter. SxS just unnecessarily complicates things in this case.

Reply:
Microsoft products usually use `(x86)` and `(x64)`. May be we can do it the Microsoft way? 😊

![image](https://github.com/Azure/azure-cli/assets/4003950/b48997e5-e9e6-4cd6-be6f-ee37435eda78)